test/hotspot/jtreg/compiler/loopopts/superword/TestMemorySegment.java

```diff
@@ -0,0 +1,860 @@
+
+            if (mg.byteSize() != mr.byteSize()) {
+                throw new RuntimeException("verify " + name + ": MemorySegment must have same byteSize:" +
+                                       " gold[" + i + "].byteSize = " + mg.byteSize() +
+                                       " result[" + i + "].byteSize = " + mr.byteSize());
+            }
+
+            for (int j = 0; j < (int)mg.byteSize(); j++) {
+                byte vg = mg.get(ValueLayout.JAVA_BYTE, j);
+                byte vr = mr.get(ValueLayout.JAVA_BYTE, j);
+                if (vg != vr) {
+                    throw new RuntimeException("verify " + name + ": MemorySegment must have same content:" +
+                                               " gold[" + i + "][" + j + "] = " + vg +
+                                               " result[" + i + "][" + j + "] = " + vr);
+                }
+            }
+        }
+    }
+
+    @Run(test = {"testMemorySegmentBadExitCheck",
+                 "testIntLoop_iv_byte",
+                 "testIntLoop_longIndex_intInvar_sameAdr_byte",
+                 "testIntLoop_longIndex_longInvar_sameAdr_byte",
+                 "testIntLoop_longIndex_intInvar_byte",
+                 "testIntLoop_longIndex_longInvar_byte",
+                 "testIntLoop_intIndex_intInvar_byte",
+                 "testIntLoop_iv_int",
+                 "testIntLoop_longIndex_intInvar_sameAdr_int",
+                 "testIntLoop_longIndex_longInvar_sameAdr_int",
+                 "testIntLoop_longIndex_intInvar_int",
+                 "testIntLoop_longIndex_longInvar_int",
+                 "testIntLoop_intIndex_intInvar_int",
+                 "testLongLoop_iv_byte",
+                 "testLongLoop_longIndex_intInvar_sameAdr_byte",
+                 "testLongLoop_longIndex_longInvar_sameAdr_byte",
+                 "testLongLoop_longIndex_intInvar_byte",
+                 "testLongLoop_longIndex_longInvar_byte",
+                 "testLongLoop_intIndex_intInvar_byte",
+                 "testLongLoop_iv_int",
+                 "testLongLoop_longIndex_intInvar_sameAdr_int",
+                 "testLongLoop_longIndex_longInvar_sameAdr_int",
+                 "testLongLoop_longIndex_intInvar_int",
+                 "testLongLoop_longIndex_longInvar_int",
+                 "testLongLoop_intIndex_intInvar_int"})
+    void runTests() {
+        for (Map.Entry<String,TestFunction> entry : tests.entrySet()) {
+            String name = entry.getKey();
+            TestFunction test = entry.getValue();
+            // Recall gold value from before compilation
+            Object[] gold = golds.get(name);
+            // Compute new result
+            Object[] result = test.run();
+            // Compare gold and new result
+            verify(name, gold, result);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_VECTOR_B, "= 0",
+                  IRNode.ADD_VB,        "= 0",
+                  IRNode.STORE_VECTOR,  "= 0"},
```

Comment:
You should use `failOn` instead of `= 0`. Same for other tests.

Reply:
I honestly prefer "= 0", because it is easier to flip to "> 0", and keeps the same style that way. But I guess that is really a matter of taste.